src/status_im/chat/views/message/message.cljs

```diff
@@ -257,12 +257,14 @@
   (letsubs [{:keys [photo-path]} [:get-current-account]]
     (photo from photo-path)))
 
-(defview message-author-name [from]
+(defview message-author-name [from message-username]
   (letsubs [username    [:contact-name-by-identity from]]
-    [react/text {:style style/message-author-name} username]))
+    [react/text {:style style/message-author-name} (or username
+                                                       message-username
+                                                       (gfycat/generate-gfy from))]))
```

Comment:
@cammellos `(gfycat/generate-gfy)` would be potentially called for each message row in group chat. It takes 1-3ms for me, not much. Do you think we should call it many times for same username?

Reply:
Or add to app-db? Or memoize?

Reply:
@dmitryn i believe we have username for group chat messages, so we don't need this `generate-gfy` here at all, but yeah we should be careful with messages

Reply:
@flexsurfer if we don't need it, then that call should be removed, right? :)

Reply:
Is there such a thing as messages without a username? we could enforce that to be always present when sending (generating if not there), so we could just remove the fallback. What do you think?

Reply:
But it should not be generated on-the-fly for each message, doesn't make sense to me.

Reply:
@dmitryn I agree, if we though make sure is always sent as a part of the message, that code can be just removed. Is there any reason why we should be sending a message without a username (can be the generated one if the user has not set one)?

Reply:
@cammellos message always has `from` attrubute, but not username afaik. Adding username could increase payload size. Do you guys plan to add `username` to message in new protocol branch?

Reply:
we should rethink all these things related to contacts, i think we should add every "unknown" contact in contacts with pending? flag, and generate fields only once, in that case we don't need all this magic in profile etc, and we shouldn't send name in every message for sure @yenda @janherich , we should directly request name and photo, and do not propagate photo through all network

Reply:
It also depend on when you add username. You could add dynamically once you receve message and put it to app-db or you could add it on protocol level. I would choose first variant.

Reply:
For this quick PR maybe drop the gfy part? Then revisit once new-protocol is there?

Reply:
@cammellos but for now lets just use cache for `generate-gfy`

Reply:
Yes, i think we could drop it. Message should already have username once it stored in app-db. No need to generate it on-the-fly.

Reply:
seems like the majority is for dropping, @flexsurfer are you ok with that or you feel strongly about memoizing/caching it?

Reply:
Using the new app flow a username is always set, so should not be a problem

Reply:
lets drop for this PR

Reply:
i mean lets keep this PR without changes ,if we always have username this generate-gfy will be called only in rare cases

Reply:
@flexsurfer great timing, I was just about to push enter :)

Reply:
@cammellos Please add a comment that this should be revisited after the new protocol